vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/ApolloWSMessage.java

```diff
@@ -43,4 +44,21 @@
    */
   JsonObject content();
 
+  /**
+   * @return the connection params
+   */
+  Object connectionParams();
+
+  /**
+   * Apollo handler will reply to user only when provided future completes.
+   * Handshake can only be set for message of type {@link ApolloWSMessageType#CONNECTION_INIT}.
+   * Resulting {@link Object} will be available using {@link ApolloWSMessage#connectionParams()} method
+   * on all subsequent messages of type {@link ApolloWSMessageType#START}.
+   */
+  void setHandshake(Future<Object> future);
```

Comment:
Just my 2cts, shouldn't this use a `Authentication/AuthorizationProvider` instead? this way support for JWT or other kind of auth is provided "out of the box" to the user?

Reply:
This would be ideal, will also look into ways of integrating it like this

Reply:
@pmlopes @kbobrowski what would that buy us? We don't know if users will want to verify identity, authentication, both or something else. And they might not be using `vertx-auth` at all.

However if we stick to this generic mechanism, users of `vertx-auth` can provide the future returned by, for example, `authnProvider.authenticate()`.

Reply:
@tsegismont good point, implementation would be cleaner if we keep just current approach. It may feel a bit strange for the user (as auth handler would have to go after Apollo handler, and JWT auth would need to be handled in two places), but perhaps with a good example it won't be a problem.